### ISSUE TITLE
Support for specifying queries in configuration

### ISSUE THREAD

**Issue by aksh1618** [mizlan/iswap.nvim#91]
Hey there, thanks for this awesome plugin!

I had one small suggestion that I wanted to discuss (creating an issue as discussions aren't enabled for this repository): I was wondering if support could be added for specifying queries in the configuration itself, maybe in a map, instead of just in the plugin's queries directory.

This was motivated by my use case wherein I just wanted to be able to swap around some values for some parameters in a css snippet for a markdown editing tool (Obsidian). As there is no query file defined for css currently, I have two options:
1. Create a query file for css in my local computer: This is what I did for now, adding just `(declaration)` to the list. However, this has two drawbacks:
    i. I'll have to ensure manually creating it on other machines where I want to use it.
    ii. It could get lost when updating the plugin (not entirely sure about how my package manager (lazy) handles such cases)
2. Create a query file for css and raise a PR: This seems too much for just one query, I would have gone for this option if I had more usage of css, but now if I wanted to go for a PR I would have to look into what other nodes someone working with css would like to swap, which I might not be very well suited to be the judge of! (P.S.: Let me know if you feel fine with minimal PRs like this for new languages)

Instead of these two methods, I would prefer being able to specify a map in the plugin configuration for languages, maybe something like:

```lua
require('iswap').setup{
    additional_queries = {
        "css" = [[
            (declaration)
        ]]
    }
}
```

(Another use case I can think of could be a custom file type, which might not be useful for others so won't make sense being included in this repository. This is however a completely hypothetical use case.)

**Comment by aksh1618** [comment]
As I was drafting this issue, I made some interesting discoveries (live and learn!):
- One can define queries in their own configuration directory ([source](https://github.com/nvim-treesitter/nvim-treesitter?tab=readme-ov-file#adding-queries)), even ensuring merging instead of overriding the queries provided by a plugin using `;;extends`. For example, for the use case above I could add: `$HOME/.config/nvim/queries/css/iswap-list.scm`:
    ```scm
    ;;extends
    [
      (declaration)
    ] @iswap-list

    ```
- One can just use `ISwapNode` for such once-in-a while use cases: it works for arbitrary nodes and indeed works quite well for the use case above!

Closing this as I no longer see any use cases warranting support for queries as part of configuration, given the above two options available in such situations. Thanks again for this awesome plugin! :heart:

**Comment by aksh1618** [comment]
As I was drafting this issue, I made some interesting discoveries (live and learn!):
- One can define queries in their own configuration directory ([source](https://github.com/nvim-treesitter/nvim-treesitter?tab=readme-ov-file#adding-queries)), even ensuring merging instead of overriding the queries provided by a plugin using `;;extends`. For example, for the use case above I could add: `$HOME/.config/nvim/queries/css/iswap-list.scm`:
    ```scm
    ;;extends
    [
      (declaration)
    ] @iswap-list

    ```
- One can just use `ISwapNode` for such once-in-a while use cases: it works for arbitrary nodes and indeed works quite well for the use case above!

Closing this as I no longer see any use cases warranting support for queries as part of configuration, given the above two options available in such situations. Thanks again for this awesome plugin! :heart:
